Tribler/community/multichain/block.py

```diff
@@ -78,68 +71,82 @@ def validate(self, database):
-                result[0] = PARTIAL_NEXT
+                # We pretend prev_blk exists. This leaves us with next missing, which means partial-next at best.
+                result[0] = ValidationResult.partial_next
         elif not prev_blk and next_blk:
-            # The previous block does not exist in the database, at best our result can now be partial w.r.t. prev
-            if self.sequence_number != GENESIS_SEQ and self.previous_hash != GENESIS_HASH:
-                # We are not checking the first block after genesis, so we are really missing the previous block
-                result[0] = PARTIAL_PREV
-                if next_blk.sequence_number != self.sequence_number + 1:
-                    # If both sides are unknown or non-contiguous return a full partial result.
-                    result[0] = PARTIAL
+            # Is this block a non genesis block?
+            if not is_genesis:
+                # We are really missing prev_blk. So now partial-prev at best.
+                result[0] = ValidationResult.partial_previous
+                if is_next_gap:
+                    # Both sides are unknown or non-contiguous return a full partial result.
+                    result[0] = ValidationResult.partial
+            elif is_next_gap:
+                # The next block does not exist in the database, at best our result can now be full partial
+                result[0] = ValidationResult.partial
         elif prev_blk and not next_blk:
-            # The next block does not exist in the database, at best our result can now be partial w.r.t. next
-            result[0] = PARTIAL_NEXT
-            if prev_blk.sequence_number != self.sequence_number - 1:
-                # If both sides are unknown or non-contiguous return a full partial result.
-                result[0] = PARTIAL
+            # We are missing next_blk, so now partial-next at best.
+            result[0] = ValidationResult.partial_next
+            if is_prev_gap:
+                # Both sides are unknown or non-contiguous return a full partial result.
+                result[0] = ValidationResult.partial
         else:
             # both sides have known blocks, see if there are gaps
-            if (prev_blk.sequence_number != self.sequence_number - 1) and \
-                    (next_blk.sequence_number != self.sequence_number + 1):
-                result[0] = PARTIAL
-            elif prev_blk.sequence_number != self.sequence_number - 1:
-                result[0] = PARTIAL_PREV
-            elif next_blk.sequence_number != self.sequence_number + 1:
-                result[0] = PARTIAL_NEXT
+            if is_prev_gap and is_next_gap:
+                result[0] = ValidationResult.partial
+            elif is_prev_gap:
+                result[0] = ValidationResult.partial_previous
+            elif is_next_gap:
+                result[0] = ValidationResult.partial_next()
```

Comment:
hehehe, I pushed accidentally. I wasn't done yet -.- This and a bunch of other small bits where all caught when runninng the unit tests.